exercises/hardware_access_control/README.md

```diff
@@ -0,0 +1,160 @@
+<!--
+Copyright lowRISC Contributors.
+SPDX-License-Identifier: Apache-2.0
+-->
+# Hardware Access Control Exercise
+
+If you haven't already, pop yourself up to the '[building the exercises][]' section to see how the exercises are built.
+
+[Building the Exercises]: ../README.md#building-the-exercises
+
+This exercise we utilise the compartmentalisation available in CHERIoT RTOS to control access to a hardware peripheral: the very security critical LEDs.
+
+For this exercise, when the [`xmake.lua`][] build file is mentioned `exercises/hardware_access_control/xmake.lua` is being referred to.
+
+[`xmake.lua`]: ../../exercises/hardware_access_control/xmake.lua
+
+## Part 1
+
+Let's start with the firmware image called `hardware_access_part_1` in the [`xmake.lua`][] file.
+This image has two threads running two compartments: `blinky_raw` and `led_walk_raw`.
+`blinky_raw` simply toggles an LED and `led_walk_raw` walks through all the LEDs toggling them as it goes.
+The sources of these compartments can be found in [`exercises/hardware_access_control/part_1/`][].
+
+[`exercises/hardware_access_control/part_1/`]: ../../exercises/hardware_access_control/part_1/
+
+Let's look inside `blinky_raw`.
+It uses the RTOS' `MMIO_CAPABILITY` macro to get the capability that grants it access to the GPIO MMIO region.
+This magic macro will handle adding the MMIO region to the compartment's imports and mapping it to a type, in this case `SonataGPIO` (from [`platform-gpio.hh`][]).
+*For more information on this macro, see [the drivers section of CHERIoT programmers guide][].*
+
+[the drivers section of CHERIoT programmers guide]: https://cheriot.org/book/top-drivers-top.html#mmio_capabilities
+[`platform-gpio.hh`]: ../../cheriot-rtos/sdk/include/platform/sunburst/platform-gpio.hh
+
+This is great!
+If you build and [load][running on fpga] the `hardware_access_part_1` firmware on the FPGA, you have flashing LEDs!
+What more could one want?
+
+[running on fpga]: ../../doc/guide/running-software.md#running-on-the-sonata-fpga
+
+Well maybe some level of access control.
+Currently both `blinky_raw` and `led_walk_raw` have access to all of the GPIO ports, and neither can trust the other compartment isn't toggling the LED when they are not looking.
+The keen eyed among you will have noticed that this is already happening with both toggling user LED 7.
```

Comment:
> The keen eyed among you will ...

Grammar: The **keen-eyed** among you will ...

Reply:
Addressed in: https://github.com/lowRISC/sonata-software/pull/42